### DISCUSSION TITLE
Call RFM with destination host through PyRFC

### DISCUSSION THREAD

**Discussion by htkarabulut** [SAP/PyRFC#335] [Q&A] [answered]
We have a SAP system that consists of 5 instances. We are using the RSAU_READ_LOG function module for fetching the SAL logs. We need the functionality that should provide using this RFC with `destinations` but we cannot find a way to use the PyRFC library for this purpose. Can you help me on this?

For example:
```py
result = conn.call('RSAU_READ_LOG', \
  DESTINATION = <sap instance name> # we need something like this
  # ...
```

**Comment by bsrdjan** [comment]
Can you create local sapnwrfc.Ini file, define destinations there and open client connections like.

client = Connection(dest="mySystem") 

See also

 - https://help.sap.com/docs/SAP_NETWEAVER_750/753088fc00704d0a80e7fbd6803c8adb/48ce50e418d3424be10000000a421937.html?state=DRAFT&version=7.5.4
- https://github.com/SAP/PyRFC/issues/241

**Reply by htkarabulut** [reply, marked as answer]
Hmm, actually we are developing an automated pentesting toolchain and unfortunately I prefer not to access local files.
I am currently handling connections by getting the system parameters from a database. A web UI interacts with the database to save/edit the system parameters. 

However, I think I will just go with the way of retrieving the systems with `TH_SERVER_LIST` then calling `RSAU_READ_LOG` with each of the servers by creating a connection from just changing the `ASHOST` and `SYSNR` (the last 2 chars in the name of the instance).
